### ISSUE TITLE
Development environment

### ISSUE THREAD

**Issue by anfiel** [osrf/subt_hello_world#16]
[+] Building 31.7s (2/2) FINISHED                                docker:default
 => [internal] load build definition from Dockerfile                       0.0s
 => => transferring dockerfile: 4.35kB                                     0.0s
 => ERROR [internal] load metadata for docker.io/nvidia/cudagl:11.4.2-de  31.7s
------
 > [internal] load metadata for docker.io/nvidia/cudagl:11.4.2-devel-ubuntu18.04:
------
Dockerfile:12
--------------------
  10 |     # https://hub.docker.com/r/nvidia/cudagl/
  11 |     # https://github.com/NVIDIA/nvidia-docker/wiki/CUDA
  12 | >>> FROM nvidia/cudagl:11.4.2-devel-ubuntu18.04
  13 |     
  14 |     # setup timezone
--------------------
ERROR: failed to solve: nvidia/cudagl:11.4.2-devel-ubuntu18.04: failed to resolve source metadata for docker.io/nvidia/cudagl:11.4.2-devel-ubuntu18.04: failed to copy: httpReadSeeker: failed open: failed to do request: Get "https://production.cloudflare.docker.com/registry-v2/docker/registry/v2/blobs/sha256/bc/bc4eea507469a745615c69a35a7f1c14978366be1066714aadfbb834be098bbf/data?verify=1718679139-aE%2Bztz17t1cRMJKtbuHvhJrzhZo%3D": dial tcp [2a03:2880:f10e:83:face:b00c:0:25de]:443: i/o timeout

**Comment by peci1** [comment]
Please, provide more context and describe what you tried to do. The report is not useful in its current state and there's nothing we can do for you unless you provide more information.

**Comment by anfiel** [comment]
![2024-06-18 15-49-57屏幕截图](https://github.com/osrf/subt_hello_world/assets/142065816/22b54dc8-346d-43bc-9a65-d9a2db4a8572)
My host computer is ubuntu20.04 and cuda is 11.8. When I use this command, the above error is displayed. When I change the cuda version in dockerfile, the error is still displayed. Why is this?

**Comment by peci1** [comment]
Thanks for the update. Could you check if #17 resolves the issue for you?

**Comment by anfiel** [comment]
![2024-06-19 09-55-57屏幕截图](https://github.com/osrf/subt_hello_world/assets/142065816/10680d9d-f0eb-4b2e-b440-1b687cae0ee5)
It seems like the same, maybe Docker.io/nvidia/cudagl:10.1-devel-ubuntu18.04. Is there a problem here?

**Comment by peci1** [comment]
No, there's not in general. I succeeded building it from scratch yesterday. Can you try over a VPN or a different ISP's network?

Or, as I see you access it via IPv6, try disabling IPv6 in your network connection settings.

**Comment by anfiel** [comment]
Every time ./build_common_image.bash is run, an error is reported when it reaches the 1178th line, as shown in the following figure
![2024-06-21 16-04-25屏幕截图](https://github.com/osrf/subt_hello_world/assets/142065816/f6c87b69-1d2d-424a-9509-70ed85828c62)
![2024-06-21 16-01-35屏幕截图](https://github.com/osrf/subt_hello_world/assets/142065816/f5835dcd-745d-4f26-b06c-86915ff1fbff)

**Comment by peci1** [comment]
Your networking seems wrong. Look at the `Bad Gateway [IP 127.0.0.1]`. Not sure what does this, but it is almost 100% wrong. Check your `/etc/hosts`.

**Comment by anfiel** [comment]
Ok, thanks for your reply, I'll check it carefully.
